File: app/src/main/java/com/example/resapp/MenuItemActivity.java
package com.example.resapp;

import android.database.Cursor;
import android.os.Bundle;
import android.text.TextUtils;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.bumptech.glide.Glide;

public class MenuItemActivity extends AppCompatActivity {

    private ImageView imgLarge;
    private TextView tvName, tvIngredients;
    private Button btnBack;

    private DatabaseHelper dbHelper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_menu_item);

        imgLarge = findViewById(R.id.imgLarge);
        tvName = findViewById(R.id.tvItemName);
        tvIngredients = findViewById(R.id.tvIngredients);
        btnBack = findViewById(R.id.btnBackToMenu);

        dbHelper = new DatabaseHelper(this);

        int menuItemId = getIntent().getIntExtra("menuItemId", -1);
        if (menuItemId != -1) {
            loadMenuItem(menuItemId);
        }

        btnBack.setOnClickListener(v -> finish());
    }

    private void loadMenuItem(int id) {
        Cursor cursor = null;

        try {
            cursor = dbHelper.getMenuItemById(id);

            if (cursor != null && cursor.moveToFirst()) {
                String name = cursor.getString(cursor.getColumnIndexOrThrow(DatabaseHelper.MENU_NAME));
                double price = cursor.getDouble(cursor.getColumnIndexOrThrow(DatabaseHelper.MENU_PRICE));
                String allergens = cursor.getString(cursor.getColumnIndexOrThrow(DatabaseHelper.MENU_ALLERGENS));

                String imageUrl = cursor.getString(cursor.getColumnIndexOrThrow(DatabaseHelper.MENU_IMAGE));

                tvName.setText(name + " - £" + price);
                tvIngredients.setText("Allergens: " + allergens);

                Glide.with(this)
                        .load(TextUtils.isEmpty(imageUrl) ? null : imageUrl)
                        .placeholder(R.drawable.burger)
                        .error(R.drawable.burger)
                        .into(imgLarge);
            }

        } finally {
            if (cursor != null) cursor.close();
        }
    }
}

File: app/src/main/java/com/example/resapp/MenuItemActivity.kt
package com.example.resapp

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MenuItemActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_menu_item)

        val img = findViewById<ImageView>(R.id.imgLarge)
        val name = findViewById<TextView>(R.id.tvItemName)
        val ingredients = findViewById<TextView>(R.id.tvIngredients)
        val allergens = findViewById<TextView>(R.id.tvAllergens)
        val backButton = findViewById<Button>(R.id.btnBackToMenu)

        img.setImageResource(intent.getIntExtra("image", 0))
        name.text = intent.getStringExtra("name")
        ingredients.text = "Ingredients: " + intent.getStringExtra("ingredients")
        allergens.text = "Allergens: " + intent.getStringExtra("allergens")

        backButton.setOnClickListener {
            finish()
        }
    }
}
